File: javascript/src/pairwise.js
/**
* Transform a list into pairwise sums of contiguous entries:
*
* Input: A list of numbers. An example is [1, 9, 2, 4, 1, 4].
* Output: The pairwise sums of the elements. For the given example, 
* the return value is a list containing [10, 11, 6, 5, 5].
* @function getPairwiseSums
* @param {Array} nums one dimensional list of numerical values. 
* An example is [1, 9, 2, 4, 1, 4]
* @yield {Array} A list containing the pairwise sum of the input. For the 
* given example, the return value is a list containing [10, 11, 6, 5, 5].
*/

function* getPairwiseSums(nums){
    const sums=[];
    for (let i = 0; i < nums.length-1; i++) {
        sums.push(nums[i]+nums[i+1]);        
    }
    yield sums;
}


module.exports = getPairwiseSums

File: python/generators/pairwise.py
def getPairwiseSums(nums):
    """ Transforms a list into pairwise sums of contiguous entries:

        Parameters
        ----------
        nums : list
            A 1 dimensional list of numbers
    """
    
    pairwiseSums = [] #declare a list to hold the sums of the consecutive list items
    #[:-1] returns all the elements of the list but the last one
    for i, _ in enumerate(nums[:-1]):  #use enumerate to create an indexed pair list and iterate
                                         # through that list to get the sums of the consecutive numbers
        pairwiseSums.append(nums[i] + nums[i+1]) #append the sum of the 2 consecutive numbers to the pairwiseSums list
    yield pairwiseSums

sums = getPairwiseSums([1,9,2,4,1,4])
print(next(sums))